fastapi/_compat.py

```diff
@@ -136,18 +150,42 @@ def serialize(
             exclude_defaults: bool = False,
             exclude_none: bool = False,
         ) -> Any:
-            # What calls this code passes a value that already called
-            # self._type_adapter.validate_python(value)
-            return self._type_adapter.dump_python(
-                value,
-                mode=mode,
-                include=include,
-                exclude=exclude,
-                by_alias=by_alias,
-                exclude_unset=exclude_unset,
-                exclude_defaults=exclude_defaults,
-                exclude_none=exclude_none,
-            )
+            try:
+                # What calls this code passes a value that already called
+                # self._type_adapter.validate_python(value)
+                return self._type_adapter.dump_python(
+                    value,
+                    mode=mode,
+                    include=include,
+                    exclude=exclude,
+                    by_alias=by_alias,
+                    exclude_unset=exclude_unset,
+                    exclude_defaults=exclude_defaults,
+                    exclude_none=exclude_none,
+                )
+            except AttributeError:
+                # pydantic v1
+                try:
+                    return value.dict(
+                        include=include,
+                        exclude=exclude,
+                        by_alias=by_alias,
+                        exclude_unset=exclude_unset,
+                        exclude_defaults=exclude_defaults,
+                        exclude_none=exclude_none,
+                    )
+                except AttributeError:
+                    return [
```

Comment:
I am not sure if there is a better way to handle the case where there is a container of model instances.

Reply:
`isinstance()`?

```python
if isinstance(value, dict):
    ...
```

Reply:
At least for Python 3.11+, I expect its zero-overhead exception handling to be a bit more efficient. In case that's neglectable, an if/else branch with `isinstance` might be more readable.